File: medium/1855-maximum-distance-between-a-pair-of-values/solution.java
/**
 * 1855. Maximum Distance Between a Pair of Values
 * Time: O(n + m)
 * Space: O(1)
 */
class Solution {
    public int maxDistance(int[] nums1, int[] nums2) {
        int n1 = nums1.length, n2 = nums2.length;
        int i = 0, j = 0;
        int maxDist = 0;

        while (i < n1 && j < n2) {
            if (nums1[i] <= nums2[j]) {
                maxDist = Math.max(maxDist, j - i);
                j++;
            } else {
                i++;
            }
        }

        return maxDist;
    }
}

File: medium/1855-maximum-distance-between-a-pair-of-values/solution.py
from typing import List


class Solution:
    def maxDistance(self, nums1: List[int], nums2: List[int]) -> int:
        """
        1855. Maximum Distance Between a Pair of Values
        Time: O(n + m)
        Space: O(1)
        """
        n1, n2 = len(nums1), len(nums2)
        i, j = 0, 0
        max_dist = 0

        while i < n1 and j < n2:
            if nums1[i] <= nums2[j]:
                max_dist = max(max_dist, j - i)
                j += 1
            else:
                i += 1

        return max_dist
